### DISCUSSION TITLE
Unpivoting stats and attributes?

### DISCUSSION THREAD

**Discussion by tylerjamesf** [tylerjamesf/MissingNo#6] [Ideas]
> One thing I would potentially consider changing is unpivoting the stats (i.e. add new records vs columns for any new xyz attributes that needs adjusted), because I love normalizing the sh!t out of a db, but i'd understand that would be more complicated to handle. So I am not mad about this implementation! <3
@evokercodex

I'm not sure what this would mean? Could you explain?

**Comment by evokercodex** [comment]
Another way to potentially view the [List of Natures](https://bulbapedia.bulbagarden.net/wiki/Nature) table might look like below. This way you add more records vs having to add more attributes / objects. More normalized and could help be extended but requires more work to flatten and apply in other areas.

**Nature**
NatureID | Label
0 | Hardy
1 | Loney
2 | Brave

**Effect**
EffectID | Label
1 | Attack
2 | Defense
3 | Spicy

**NatureEffectXref**
XrefID | NatureID | EffectID | Value
1 | 0 | 1 |  1.1
1 | 0 | 2 |  0.9
1 | 0 | 3 |  1.1
1 | 0 | 4 |  0.9
